File: allbeingSort.cs
using System;
using System.Collections.Generic;

class AllbeingSorter
{
    static bool IsSorted(int[] array)
    {
        for (int i = 1; i < array.Length; i++)
        {
            if (array[i] < array[i - 1])
            {
                return false;
            }
        }
        return true;
    }

    static void GeneratePermutations(int[] array, int level, List<int[]> results)
    {
        if (level == array.Length)
        {
            results.Add((int[])array.Clone());
            return;
        }

        for (int i = level; i < array.Length; i++)
        {
            Swap(ref array[level], ref array[i]);
            GeneratePermutations(array, level + 1, results);
            Swap(ref array[level], ref array[i]);
        }
    }

    static void Swap(ref int a, ref int b)
    {
        int temp = a;
        a = b;
        b = temp;
    }

    static void AllbeingSort(int[] array)
    {
        List<int[]> permutations = new List<int[]>();
        GeneratePermutations(array, 0, permutations);

        foreach (int[] permutation in permutations)
        {
            if (IsSorted(permutation))
            {
                Console.Write("Sorted array: ");
                Console.WriteLine(string.Join(" ", permutation));
                return;
            }
        }

        Console.WriteLine("Array is not sortable.");
    }

    static void Main()
    {
        // 示例用法
        int[] array = { 3, 2, 1 };

        Console.Write("Original array: ");
        Console.WriteLine(string.Join(" ", array));

        AllbeingSort(array);
    }
}

File: AllbeingSort.java
import java.util.ArrayList;
import java.util.List;

public class AllbeingSort {

    public static boolean isSorted(int[] array) {
        for (int i = 1; i < array.length; i++) {
            if (array[i] < array[i - 1]) {
                return false;
            }
        }
        return true;
    }

    public static void generatePermutations(int[] array, int level, List<int[]> results) {
        if (level == array.length) {
            results.add(array.clone());
            return;
        }

        for (int i = level; i < array.length; i++) {
            swap(array, level, i);
            generatePermutations(array, level + 1, results);
            swap(array, level, i);
        }
    }

    public static void swap(int[] array, int i, int j) {
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }

    public static void allbeingSort(int[] array) {
        List<int[]> permutations = new ArrayList<>();
        generatePermutations(array, 0, permutations);

        for (int[] permutation : permutations) {
            if (isSorted(permutation)) {
                System.out.print("Sorted array: ");
                for (int num : permutation) {
                    System.out.print(num + " ");
                }
                System.out.println();
                return;
            }
        }

        System.out.println("Array is not sortable.");
    }

    public static void main(String[] args) {
        // 示例用法
        int[] array = {3, 2, 1};

        System.out.print("Original array: ");
        for (int num : array) {
            System.out.print(num + " ");
        }
        System.out.println();

        allbeingSort(array);
    }
}
